PATCH-P0-FEATURE-UPSTREAM: Adds group option

This patch let's you specify not only user to use but also group that MySQL
should use.

Maintainer: Michal Hrusecky <mhrusecky@suse.cz>

Index: scripts/mysqld_safe.sh
===================================================================
--- scripts/mysqld_safe.sh.orig
+++ scripts/mysqld_safe.sh
@@ -23,6 +23,7 @@ logging=init
 want_syslog=0
 syslog_tag=
 user='@MYSQLD_USER@'
+group='@MYSQLD_GROUP@'
 pid_file=
 err_log=
 timestamp_format=UTC
@@ -201,6 +202,7 @@ parse_arguments() {
       --pid-file=*) pid_file="$val" ;;
       --plugin-dir=*) PLUGIN_DIR="$val" ;;
       --user=*) user="$val"; SET_USER=1 ;;
+      --group=*) group="$val"; SET_USER=1 ;;
 
       # these might have been set in a [mysqld_safe] section of my.cnf
       # they are added to mysqld command line to override settings from my.cnf
@@ -618,11 +620,17 @@ then
   if test "$user" != "root" -o $SET_USER = 1
   then
     USER_OPTION="--user=$user"
+    GROUP_OPTION="--group=$group"
   fi
   # Change the err log to the right user, if it is in use
   if [ $want_syslog -eq 0 ]; then
     touch "$err_log"
-    chown $user "$err_log"
+    if [ "$user" -a "$group" ]; then
+      chown $user:$group $err_log
+    else
+      [ "$user" ] && chown $user $err_log
+      [ "$group" ] && chgrp $group $err_log
+    fi
   fi
   if test -n "$open_files"
   then
@@ -641,7 +649,12 @@ mysql_unix_port_dir=`dirname $safe_mysql
 if [ ! -d $mysql_unix_port_dir ]
 then
   mkdir $mysql_unix_port_dir
-  chown $user $mysql_unix_port_dir
+  if [ "$user" -a "$group" ]; then
+    chown $user:$group $mysql_unix_port_dir
+  else
+    [ "$user" ] && chown $user $mysql_unix_port_dir
+    [ "$group" ] && chgrp $group $mysql_unix_port_dir
+  fi
   chmod 755 $mysql_unix_port_dir
 fi
 
Index: scripts/CMakeLists.txt
===================================================================
--- scripts/CMakeLists.txt.orig
+++ scripts/CMakeLists.txt
@@ -280,6 +280,7 @@ ENDIF()
 
 SET(HOSTNAME "hostname")
 SET(MYSQLD_USER "mysql")
+SET(MYSQLD_GROUP "mysql")
 
 ENDIF(UNIX)
 
Index: support-files/CMakeLists.txt
===================================================================
--- support-files/CMakeLists.txt.orig
+++ support-files/CMakeLists.txt
@@ -29,6 +29,7 @@ ELSE()
   SET(CFLAGS ${CMAKE_C_FLAGS})
   SET(CXXFLAGS ${CMAKE_CXX_FLAGS})
   SET(MYSQLD_USER "mysql")
+  SET(MYSQLD_GROUP "mysql")
   SET(ini_file_extension "cnf")
   SET(HOSTNAME "hostname")
   SET(CNF_SOCKET_LINE "# socket = .....")
